modules/library/metadata/src/main/java/org/geotools/factory/GeoTools.java

```diff
@@ -786,7 +791,69 @@ public static Hints addDefaultHints(final Hints hints) {
         }
         return completed;
     }
-    
+
+    /**
+     * Returns the default entity resolver, used to configure {@link SAXParser}.
+     * 
+     * @param hints An optional set of hints, or {@code null} if none, see {@link Hints#ENTITY_RESOLVER}.
+     * @return An entity resolver (never {@code null})
+     */
+    public static EntityResolver getEntityResolver(Hints hints) {
+        if (hints != null && hints.containsKey(Hints.ENTITY_RESOLVER)) {
+            Object resolver = hints.get(Hints.ENTITY_RESOLVER);
+
+            if (resolver == null) { // use null instance rather than check each time
+                return NullEntityResolver.INSTANCE;
+            }
+            else if (resolver instanceof EntityResolver){
+                return (EntityResolver) resolver;
+            }
+        }
+        Hints defaults = GeoTools.getDefaultHints();
+        if (defaults.containsKey(Hints.ENTITY_RESOLVER)) {
+            Object hint = defaults.get(Hints.ENTITY_RESOLVER);
+            if (hint instanceof EntityResolver) {
+                return (EntityResolver) hint;
+            }
+            if (hint instanceof String) {
+                final Logger LOGGER = Logging.getLogger("org.geotools.GeoTools");
+                try {
+                    String className = (String) hint;
+                    Class<?> type = Class.forName(className);
+
+                    for (Field field : type.getDeclaredFields()) {
+                        int modifier = field.getModifiers();
+                        if ("INSTANCE".equals(field.getName()) && Modifier.isStatic(modifier)
+                                && Modifier.isPublic(modifier)) {
+                            try {
+                                Object value = field.get(null);
+                                if (value != null && value instanceof EntityResolver) {
+                                    return (EntityResolver) value;
+                                }
+                            } catch (Throwable t) {
+                                LOGGER.log(Level.FINER, "Unable to instantiate ENTITY_RESOLVER: "
+                                        + className + ".INSTANCE", t);
+                                return PreventLocalEntityResolver.INSTANCE;
```

Comment:
While gotos are evil, wouldn't a labeled `break;` be better here so that we do not have multiple points from where we could be returing the default entity resolver?